cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,702 @@
+    )
+endif()
+
+# Image files, those should be queried from actual image.
+# Add the dynamic partition as an image partition.
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_HEX_FILE
+#  ${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
+#  )
+#
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_TARGET
+#  ${logical_target_for_zephyr_elf}
+#  )
+
+# Prepare the input_files, header_files, and images lists
+set(generated_path include/generated)
+foreach (image ${IMAGES})
+  # Special handling of `app_image` as this must be added as `:app` for historic reasons.
+  # `:app` is handled below.
+  if(NOT "${app_name}" STREQUAL "${image}")
+    list(APPEND prefixed_images ${DOMAIN}:${image})
+    list(APPEND images ${image})
+  endif()
+
+  sysbuild_get(${image}_input_files IMAGE ${image} VAR PM_YML_FILES CACHE)
+  sysbuild_get(${image}_binary_dir  IMAGE ${image} VAR ZEPHYR_BINARY_DIR CACHE)
+#  get_shared(${image}_input_files IMAGE ${image} PROPERTY PM_YML_FILES)
+#  get_shared(${image}_binary_dir  IMAGE ${image} PROPERTY ZEPHYR_BINARY_DIR)
+
+  list(APPEND input_files  ${${image}_input_files})
+  list(APPEND header_files ${${image}_binary_dir}/${generated_path}/pm_config.h)
+  print(header_files)
+  # Re-configure (Re-execute all CMakeLists.txt code) when original
+  # (not preprocessed) configuration file changes.
+#  get_shared(dependencies IMAGE ${image} PROPERTY PM_YML_DEP_FILES)
+#  set_property(
+#    DIRECTORY APPEND PROPERTY
+#    CMAKE_CONFIGURE_DEPENDS
+#    ${dependencies}
+#    )
+endforeach()
+
+# Explicitly add the dynamic partition image
+list(APPEND prefixed_images "${DOMAIN}:${dynamic_partition}")
+list(APPEND images ${dynamic_partition})
+print(PROJECT_BINARY_DIR)
+#list(APPEND input_files ${PROJECT_BINARY_DIR}/${generated_path}/pm.yml)
+#list(APPEND header_files ${PROJECT_BINARY_DIR}/${generated_path}/pm_config.h)
+
+# Add subsys defined pm.yml to the input_files
+list(APPEND input_files ${PM_SUBSYS_PREPROCESSED})
+
+if (DEFINED CONFIG_SOC_NRF9160)
+  # See nRF9160 Product Specification, chapter "UICR"
+  set(otp_start_addr "0xff8108")
+  set(otp_size 756) # 189 * 4
+elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
+  # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
+  set(otp_start_addr "0xff8100")
+  set(otp_size 764)  # 191 * 4
+endif()
+
+sysbuild_get(${app_name}_CONFIG_PM_SRAM_SIZE IMAGE ${app_name} VAR CONFIG_PM_SRAM_SIZE KCONFIG)
+sysbuild_get(${app_name}_CONFIG_PM_SRAM_BASE IMAGE ${app_name} VAR CONFIG_PM_SRAM_BASE KCONFIG)
+
+add_region(
+  NAME sram_primary
+  SIZE ${${app_name}_CONFIG_PM_SRAM_SIZE}
+  BASE ${${app_name}_CONFIG_PM_SRAM_BASE}
+  PLACEMENT complex
+  DYNAMIC_PARTITION sram_primary
+  )
+
+sysbuild_get(${app_name}_CONFIG_FLASH_SIZE IMAGE ${app_name} VAR CONFIG_FLASH_SIZE KCONFIG)
+math(EXPR flash_size "${${app_name}_CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
+
+if (CONFIG_SOC_NRF9160 OR CONFIG_SOC_NRF5340_CPUAPP)
```

Comment:
Above uses `DEFINED` and this doesn't, same as it is in the source file, could fix it up here

Reply:
Good point, fixed.